lxd/images.go

```diff
@@ -2637,7 +2637,12 @@ func imageValidSecret(s *state.State, r *http.Request, projectName string, finge
 			continue
 		}
 
-		if !shared.StringInSlice(fmt.Sprintf("/1.0/images/%s", fingerprint), opImages) {
+		opImagesStr := make([]string, len(opImages))
+		for i, opImage := range opImages {
+			opImagesStr[i] = opImage.String()
+		}
+
+		if !shared.StringInSlice(fmt.Sprintf("/1.0/images/%s", fingerprint), opImagesStr) {
```

Comment:
Rather than converting the URLs to a []string slice above, so you can use `shared.StringInSlice` here, it would be simpler to instead change the loop above to check for the condition here and if found set `var found bool` var to true and break the loop. Then you can add a condition to check for `!found` here.